res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -0,0 +1,262 @@
+var Prime4 = {};
+
+/*
+ * DENON DJ PRIME 4 - CONTROLLER MAPPING FOR MIXXX
+ *
+ * Thank you for using this mapping! Below are a few parameters
+ * you can change if you want to customize anything :)
+ */
+
+// What colour would you like each deck to be?
+// Choose between "red", "yellow", "green", "cyan", "blue", "magenta", or "white".
+const colourOfDeck1 = "white";
+const colourOfDeck2 = "yellow";
+const colourOfDeck3 = "cyan";
+const colourOfDeck4 = "magenta";
+
+/*
+ *                   WARNING!!!
+ *
+ *      DO NOT EDIT ANYTHING PAST THIS POINT
+ *       UNLESS YOU KNOW WHAT YOU'RE DOING.
+ */
+
+Prime4.rgbCode = {
+    black: 0,
+    blueDark: 1,
+    blueDim: 2,
+    blue: 3,
+    greenDark: 4,
+    cyanDark: 5,
+    greenDim: 8,
+    cyanDim: 10,
+    green: 12,
+    cyan: 15,
+    redDark: 16,
+    magentaDark: 17,
+    yellowDark: 20,
+    whiteDark: 21,
+    redDim: 32,
+    magentaDim: 34,
+    yellowDim: 40,
+    whiteDim: 42,
+    red: 48,
+    magenta: 51,
+    orange: 56,
+    yellow: 60,
+    white: 63,
+};
+
+const colDeck1 = eval("Prime4.rgbCode." + colourOfDeck1);
+const colDeck2 = eval("Prime4.rgbCode." + colourOfDeck2);
+const colDeck3 = eval("Prime4.rgbCode." + colourOfDeck3);
+const colDeck4 = eval("Prime4.rgbCode." + colourOfDeck4);
+const colDeck1Dark = eval("Prime4.rgbCode." + colourOfDeck1 + "Dark");
+const colDeck2Dark = eval("Prime4.rgbCode." + colourOfDeck2 + "Dark");
+const colDeck3Dark = eval("Prime4.rgbCode." + colourOfDeck3 + "Dark");
+const colDeck4Dark = eval("Prime4.rgbCode." + colourOfDeck4 + "Dark");
+
+Prime4.init = function() {
+    midi.sendShortMsg(0x90, 0x75, 0x00); // Turn off all LEDs
+    Prime4.leftDeck = new Prime4.Deck([1, 3], 4);
+    Prime4.rightDeck = new Prime4.Deck([2, 4], 5);
+    components.Button.prototype.isPress = function(channel, control, value, status) {
+        return (status & 0xF0) === 0x90;
+    };
+    midi.sendShortMsg(0x90, 0x0D, colDeck1Dark); // PFL 1
+    midi.sendShortMsg(0x91, 0x0D, colDeck2Dark); // PFL 2
+    midi.sendShortMsg(0x92, 0x0D, colDeck3Dark); // PFL 3
+    midi.sendShortMsg(0x93, 0x0D, colDeck4Dark); // PFL 4
+    midi.sendShortMsg(0x9F, 0x1C, colDeck1); // Deck 1 Toggle
+    midi.sendShortMsg(0x9F, 0x1D, colDeck2); // Deck 2 Toggle
+    midi.sendShortMsg(0x9F, 0x1E, colDeck3Dark); // Deck 3 Toggle
+    midi.sendShortMsg(0x9F, 0x1F, colDeck4Dark); // Deck 4 Toggle
+    midi.sendShortMsg(0x94, 0x21, colDeck1); // Left Jog Wheel
+    midi.sendShortMsg(0x95, 0x21, colDeck2); // Right Jog Wheel
+    midi.sendShortMsg(0x9F, 0x07, 0x01); // View Button
+};
+
+Prime4.shutdown = function() {
+    midi.sendShortMsg(0x90, 0x75, 0x01); // Return all LEDs to initial dim state
+};
+
+Prime4.Deck = function(deckNumbers, midiChannel) {
+
+    components.Deck.call(this, deckNumbers);
+
+    this.censorButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x01],
+        key: "reverseroll",
+        on: 0x7f,
+        off: 0x01,
```

Comment:
`on: 0x7F` is already the default. Instead of specifing `off: 0x01` on each component, you can overwrite it once at the top level by doing `components.Button.prototype.off = 0x01;` somewhere before the `Prime4.init` (Somwhere around line 60 is a good place).